### PR TITLE
Refactored AddingForm and changed name to CreateBookFormComponent.

### LINKED ISSUE #140: Refactor new book view
## Description:
Search and select from all titles instead of typing the exactly correct title.

## DoD:
- A user will be able to select the title of which they want to make a book of instead of having to write the name of the title.
- Possibly some kind of search or filter functionality when selecting titles.

## Context
Relates to #138

### PR DESCRIPTION
**Pull request by te4-gabriel-stark**
## What issue did you implement or fix?
Fixed #140 
Relates to #144 

## Summary
- Changed from bootstrap to vuetify in this component.
- Removed unnecessary code and rewrote it.
- Changed the file name from 'AddingForm' to 'CreateBookFormComponent' for clarity.
- Changed in App.vue so that is uses a v-app instead of a .app.
- Changed in Add.vue so that it imports the correct file.

## Type of change
- [x] New feature (non-breaking change which adds functionality)

## Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## How can this implementation be improved?
- Also update the 'RecentBook' file so it will look better on the site.

### CODE REVIEW

File: frontend/src/components/CreateBookFormComponent.vue [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+      title_id: 0,
+      status: '',
+    };
+    this.show = false;
+    this.$nextTick(() => {
+      this.show = true;
+    });
+  }
+}
+</script>
```
Comment (TE4-Mattis-Abrahamsson) [inline frontend/src/components/CreateBookFormComponent.vue]: How would you feel about adding indentation here?

File: frontend/src/components/CreateBookFormComponent.vue [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    evt.preventDefault();
+    this.form = {
+      barcode: '',
+      title_id: 0,
+      status: '',
+    };
+    this.show = false;
+    this.$nextTick(() => {
+      this.show = true;
+    });
```
Comment (TE4-Mattis-Abrahamsson) [inline frontend/src/components/CreateBookFormComponent.vue]: Why does the reset function hide the form for a short period of time?

File: frontend/src/components/CreateBookFormComponent.vue [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+  public form = {
+    barcode: '',
+    title_id: 0,
+    status: '',
+  };
+  public show = true;
+
+  get titleNames() {
+    return Object.entries(TitlesModule.all).map(([key, title]) => ({name: title.name, id: title.id}));
+  }
```
Comment (TE4-Mattis-Abrahamsson) [inline frontend/src/components/CreateBookFormComponent.vue]: Maybe we should abstract this functionality to the TitlesModule, maybe a getter called titlesArray or something

File: frontend/src/components/CreateBookFormComponent.vue (hunk 1 above)
Comment (te4-gabriel-stark) [reply frontend/src/components/CreateBookFormComponent.vue]: That sounds great!

File: frontend/src/components/CreateBookFormComponent.vue (hunk 2 above)
Comment (te4-gabriel-stark) [reply frontend/src/components/CreateBookFormComponent.vue]: I do not know, it was there before but it works without it so it seems to be useless...

File: frontend/src/components/CreateBookFormComponent.vue (hunk 3 above)
Comment (te4-gabriel-stark) [reply frontend/src/components/CreateBookFormComponent.vue]: Good idea ones again, will fix.

File: frontend/src/components/CreateBookFormComponent.vue [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+  import BooksModule from '../store/modules/BooksModule';
+
+  @Component
+  export default class CreateBookFormComponent extends Vue {
+    private form = {
+      barcode: '',
+      title_id: 0,
+      status: '',
+    };
+    private show = true;
```
Comment (TE4-Mattis-Abrahamsson) [inline frontend/src/components/CreateBookFormComponent.vue]: Is this needed?
```suggestion
```

File: frontend/src/components/CreateBookFormComponent.vue [hunk 5]
```diff
@@ -0,0 +1,59 @@
+<template>
+  <v-form @submit="onSubmit" v-if="show">
```
Comment (TE4-Mattis-Abrahamsson) [inline frontend/src/components/CreateBookFormComponent.vue]:
```suggestion
  <v-form @submit="onSubmit">
```

**Pull request merged**